pkg/manager/impl/description_entity_manager.go

```diff
@@ -0,0 +1,181 @@
+		logger.Errorf(ctx, "failed to compute description entity digest for [%+v] with err: %v", id, err)
+		return err
+	}
+
+	existingDescriptionEntityModel, err := util.GetDescriptionEntityModel(ctx, db, id)
+	if err == nil {
+		if bytes.Equal(existingDescriptionEntityModel.Digest, descriptionDigest) {
+			return errors.NewFlyteAdminErrorf(codes.AlreadyExists,
+				"identical description entity already exists with id %v", id)
+		}
+
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument,
+			"description entity with different structure already exists with id %v", id)
+	}
+
+	descriptionModel, err := transformers.CreateDescriptionEntityModel(descriptionEntity, id, descriptionDigest)
+	if err != nil {
+		logger.Errorf(ctx,
+			"Failed to transform description model [%+v] with err: %v", descriptionEntity, err)
+		return err
+	}
+
+	var descriptionID uint
+	if descriptionID, err = db.DescriptionEntityRepo().Create(ctx, descriptionModel); err != nil {
+		logger.Errorf(ctx, "Failed to create description model with id [%+v] with err %v", id, err)
+		return err
+	}
+
+	if id.ResourceType == core.ResourceType_TASK {
+		err = db.TaskRepo().UpdateDescriptionID(models.Task{
+			TaskKey: models.TaskKey{
+				Project: descriptionModel.Project,
+				Domain:  descriptionModel.Domain,
+				Name:    descriptionModel.Name,
+				Version: descriptionModel.Version,
+			},
+			DescriptionID: descriptionID,
+		})
+		if err != nil {
+			logger.Errorf(ctx, "Failed to update descriptionID in tasks table: %v", err)
+			return err
+		}
+	} else if id.ResourceType == core.ResourceType_WORKFLOW {
+		err = db.WorkflowRepo().UpdateDescriptionID(models.Workflow{
+			WorkflowKey: models.WorkflowKey{
+				Project: descriptionModel.Project,
+				Domain:  descriptionModel.Domain,
+				Name:    descriptionModel.Name,
+				Version: descriptionModel.Version,
+			},
+			DescriptionID: descriptionID,
+		})
+		if err != nil {
+			logger.Errorf(ctx, "Failed to update descriptionID in workflows table: %v", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (d *DescriptionEntityManager) GetDescriptionEntity(ctx context.Context, request admin.ObjectGetRequest) (
+	*admin.DescriptionEntity, error) {
+	if err := validation.ValidateDescriptionEntityGetRequest(request); err != nil {
+		logger.Errorf(ctx, "invalid request [%+v]: %v", request, err)
+		return nil, err
+	}
+	ctx = contextutils.WithProjectDomain(ctx, request.Id.Project, request.Id.Domain)
+	return util.GetDescriptionEntity(ctx, d.db, *request.Id)
+}
+
+func (d *DescriptionEntityManager) ListDescriptionEntity(ctx context.Context, request admin.DescriptionEntityListRequest) (*admin.DescriptionEntityList, error) {
+	// Check required fields
+	if err := validation.ValidateDescriptionEntityListRequest(request); err != nil {
+		return nil, err
+	}
+	ctx = contextutils.WithProjectDomain(ctx, request.DescriptionEntityId.Project, request.DescriptionEntityId.Domain)
+	ctx = contextutils.WithWorkflowID(ctx, request.DescriptionEntityId.Name)
```

Comment:
this isn't necessarily always the workflow name though if it's for a different entity type, right?

Reply:
fixed it